crates/polars-plan/src/dsl/mod.rs

```diff
@@ -1513,9 +1532,19 @@ impl Expr {
         }
 
         if literal_searchers {
-            self.map_many_private(FunctionExpr::Replace { return_dtype }, &args, false, false)
+            self.map_many_private(
+                FunctionExpr::ReplaceStrict { return_dtype },
+                &args,
+                false,
+                false,
+            )
         } else {
-            self.apply_many_private(FunctionExpr::Replace { return_dtype }, &args, false, false)
+            self.apply_many_private(
```

Comment:
Not for this PR, but I think we can set this to `map` later. I don't believe this is group-aware. If the input is group-aware, that would already bubble up that info.